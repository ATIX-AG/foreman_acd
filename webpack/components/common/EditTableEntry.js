import React from 'react';
import PropTypes from 'prop-types';
import { Icon, Button } from 'patternfly-react';
import { translate as __ } from 'foremanReact/common/I18n';

const EditTableEntry = ({
  hidden,
  disabled,
  handleLocking,
  onEditTableEntry,
  additionalData,
}) => {
  if (hidden) {
    return null;
  }

  let title = 'edit this entry';

  if (handleLocking === true) {
    if (additionalData.rowData.locked === true) {
      disabled = true;
      title = 'This entry is locked and can not be changed.';
    }
  }

  return (
    <span>
      <Button
        bsStyle="default"
        disabled={disabled}
        onClick={() => onEditTableEntry(additionalData)}
      >
        <Icon type="pf" name="edit" title={__(title)} />
      </Button>
    </span>
  );
};

EditTableEntry.propTypes = {
  hidden: PropTypes.bool,
  disabled: PropTypes.bool.isRequired,
  handleLocking: PropTypes.bool.isRequired,
  onEditTableEntry: PropTypes.func.isRequired,
  additionalData: PropTypes.object.isRequired,
};

export default EditTableEntry;
