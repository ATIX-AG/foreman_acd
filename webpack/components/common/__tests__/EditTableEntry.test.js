import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import EditTableEntry from '../EditTableEntry';

const noop = () => {};

const fixtures = {
  'should render edit table entry': {
    hidden: false,
    disabled: false,
    handleLocking: false,
    onEditTableEntry: noop,
    additionalData: {},
  },
  'should render hidden button': {
    hidden: true,
    disabled: false,
    handleLocking: false,
    onEditTableEntry: noop,
    additionalData: {},
  },
  'should render disabled button': {
    hidden: false,
    disabled: true,
    handleLocking: false,
    onEditTableEntry: noop,
    additionalData: {},
  },
  'should render locked button': {
    hidden: false,
    disabled: false,
    handleLocking: true,
    onEditTableEntry: noop,
    additionalData: { rowData: { locked: true } },
  },
  'should render locked and disabled button': {
    hidden: false,
    disabled: true,
    handleLocking: true,
    onEditTableEntry: noop,
    additionalData: { rowData: { locked: true } },
  },
  'should render hidde, locked and disabled button': {
    hidden: true,
    disabled: true,
    handleLocking: true,
    onEditTableEntry: noop,
    additionalData: { rowData: { locked: true } },
  },
};

describe('EditTableEntry', () =>
  testComponentSnapshotsWithFixtures(EditTableEntry, fixtures));
