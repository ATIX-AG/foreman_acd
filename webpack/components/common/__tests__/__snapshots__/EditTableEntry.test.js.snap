// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTableEntry should render disabled button 1`] = `
<span>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={true}
    onClick={[Function]}
  >
    <Icon
      name="edit"
      title="edit this entry"
      type="pf"
    />
  </Button>
</span>
`;

exports[`EditTableEntry should render edit table entry 1`] = `
<span>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    onClick={[Function]}
  >
    <Icon
      name="edit"
      title="edit this entry"
      type="pf"
    />
  </Button>
</span>
`;

exports[`EditTableEntry should render hidde, locked and disabled button 1`] = `""`;

exports[`EditTableEntry should render hidden button 1`] = `""`;

exports[`EditTableEntry should render locked and disabled button 1`] = `
<span>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={true}
    onClick={[Function]}
  >
    <Icon
      name="edit"
      title="This entry is locked and can not be changed."
      type="pf"
    />
  </Button>
</span>
`;

exports[`EditTableEntry should render locked button 1`] = `
<span>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={true}
    onClick={[Function]}
  >
    <Icon
      name="edit"
      title="This entry is locked and can not be changed."
      type="pf"
    />
  </Button>
</span>
`;
